roofit/roofitcore/src/RooAbsMinimizerFcn.h

```diff
@@ -118,6 +119,10 @@ class RooAbsMinimizerFcn {
    mutable double _funcOffset{0.};
    mutable int _numBadNLL = 0;
    mutable int _evalCounter{0};
+   // PB: these mutables signal a suboptimal design. A separate error handling
```

Comment:
Nice observation! You're planning to do this in the future?

Reply:
For me it currently looks like a nice-to-have. Factoring it out would make the RooAbsMinimizerFcn simpler and better maintainable. For the MinuitFcnGrad, it would enable easier synchronization of these error-handling related parameters between calculators (likelihood and gradient) and also easier synchronization between master and workers in the parallel calculators. The latter is currently in fact not being done at all, i.e. evalCounter does not count all the evaluations done on workers. However, I don't know how many people actually use the evalCounter or numBadNLL, so it's not clear whether this is a need-to-have.

So, no, not planned from my side, not high enough priority for me (according to what I know currently at least). I would love to help out someone who does have time for it.